src/wrap.rs

```diff
@@ -0,0 +1,35 @@
+use crate::{prelude::Error, resource::config::RootsConfig};
+
+mod gogdl;
+mod legendary;
+
+/// Trait for command line argument parsers to determine the actual game name,
+/// for implementations check the submodules.
+trait LaunchParser {
+    /// Determine game name from `commands`, return `None` if it fails.
+    fn parse(&self, roots: &[RootsConfig], commands: &[String]) -> Option<String>;
+}
+
+pub fn get_game_name_from_heroic_launch_commands(roots: &[RootsConfig], commands: &[String]) -> Result<String, Error> {
+    // I'd love to write let d = vec![Heroic{}, Legendary{}];
+    //
+    // Coming from OOP the code below seems a bit much of syntactical noise, but
+    // it handles the fact that a trait is a compile time structure with unknown
+    // size, so I "Box" it to put the actual objects on the heap.
```

Comment:
Much cleaner, thanks for this one!  I'll keep them in their module for now since I am not sure how far I have to go with each one.  Amazon Games are supported in heroic as well and ATM having it in separate files helps me not becoming confused ....

Reply:
I think in the final review of this PR we will revisit this topic again anyways.